stable/concourse/templates/web-deployment.yaml

```diff
@@ -97,255 +86,813 @@ spec:
+            {{- if .Values.concourse.web.awsSecretsManager.teamSecretTemplate }}
+            - name: CONCOURSE_AWS_SECRETSMANAGER_TEAM_SECRET_TEMPLATE
+              value: {{ .Values.concourse.web.awsSecretsManager.teamSecretTemplate | quote }}
+            {{- end }}
+            {{- end }}
+
+
+            {{- if .Values.concourse.web.awsSsm.enabled }}
+            - name: CONCOURSE_AWS_SSM_ACCESS_KEY
               valueFrom:
                 secretKeyRef:
                   name: {{ template "concourse.concourse.fullname" . }}
-                  key: gitlab-auth-client-secret
-            - name: CONCOURSE_GITLAB_AUTH_GROUP
-              value: {{ .Values.concourse.gitlabAuth.group | quote }}
-            - name: CONCOURSE_GITLAB_AUTH_AUTH_URL
-              value: {{ .Values.concourse.gitlabAuth.authUrl | quote }}
-            - name: CONCOURSE_GITLAB_AUTH_TOKEN_URL
-              value: {{ .Values.concourse.gitlabAuth.tokenUrl | quote }}
-            - name: CONCOURSE_GITLAB_AUTH_API_URL
-              value: {{ .Values.concourse.gitlabAuth.apiUrl | quote }}
-            {{- end }}
-            {{- if .Values.concourse.genericOauth.enabled }}
-            - name: CONCOURSE_GENERIC_OAUTH_CLIENT_ID
+                  key: aws-ssm-access-key
+            - name: CONCOURSE_AWS_SSM_SECRET_KEY
               valueFrom:
                 secretKeyRef:
                   name: {{ template "concourse.concourse.fullname" . }}
-                  key: generic-oauth-client-id
-            - name: CONCOURSE_GENERIC_OAUTH_CLIENT_SECRET
+                  key: aws-ssm-secret-key
+            {{- if .Values.secrets.awsSsmSessionToken }}
+            - name: CONCOURSE_AWS_SSM_SESSION_TOKEN
               valueFrom:
                 secretKeyRef:
                   name: {{ template "concourse.concourse.fullname" . }}
-                  key: generic-oauth-client-secret
-            - name: CONCOURSE_GENERIC_OAUTH_DISPLAY_NAME
-              value: {{ .Values.concourse.genericOauth.displayName | quote }}
-            - name: CONCOURSE_GENERIC_OAUTH_AUTH_URL
-              value: {{ .Values.concourse.genericOauth.authUrl | quote }}
-            - name: CONCOURSE_GENERIC_OAUTH_AUTH_URL_PARAM
-              value: {{ .Values.concourse.genericOauth.authUrlParam | quote }}
-            - name: CONCOURSE_GENERIC_OAUTH_SCOPE
-              value: {{ .Values.concourse.genericOauth.scope | quote }}
-            - name: CONCOURSE_GENERIC_OAUTH_TOKEN_URL
-              value: {{ .Values.concourse.genericOauth.tokenUrl | quote }}
-            {{- end }}
-            {{- if .Values.credentialManager.kubernetes.enabled }}
-            - name: CONCOURSE_KUBERNETES_NAMESPACE_PREFIX
-              value: {{ template "concourse.namespacePrefix" . }}
+                  key: aws-ssm-session-token
             {{- end }}
-            {{- if .Values.credentialManager.vault.enabled }}
+            {{- if .Values.concoures.web.awsSsm.region }}
+            - name: AWS_REGION
```

Comment:
Would anyone use both SSM and Secretsmanager, each in a different region? If so, that edge case isn't possible.

Reply:
do you mean `AWS_REGION` is duplicated in both ssm and secretmanager? Actually one couldn't even set this env var to different value.